ui/js/directives/treeherder/clonejobs.js

```diff
@@ -270,21 +270,12 @@ treeherder.directive('thCloneJobs', [
          * display.
          */
         var addGroupJobsAndCounts = function(jgObj, platformGroup) {
-            var ct, job, jobCountBtn, l;
-            var countAdded = false;
             var jobCountBtnArray = [];
             var jobBtnArray = [];
             var stateCounts = {};
             var lastJobSelected = ThResultSetStore.getSelectedJob($rootScope.repoName);
 
-            var jobList = platformGroup.find(".group-job-list");
-            var countList = platformGroup.find(".group-count-list");
-            jobList.empty();
-            countList.empty();
-
-            for (l = 0; l < jgObj.jobs.length; l++) {
-
-                job = jgObj.jobs[l];
+            _.forEach(jgObj.jobs, function(job) {
```

Comment:
So I did some profiling, the vast majority of the time here is spent in jQuery (see https://bugzilla.mozilla.org/show_bug.cgi?id=1164266), changing the iteration strategy won't change that. This change is more for clarity than speed.